lib/validate.js

```diff
@@ -0,0 +1,64 @@
+const _ = require('lodash');
+const u = require('url');
+
+const isValidUrl = url => (url.protocol != null) &&
+  url.protocol.match(/^http[s]?:/) &&
+  url.slashes &&
+  (url.hostname != null);
+
+
+const url = function(vars) {
+  // validate URL
+  if (vars.url == null) {
```

Comment:
this will be a string, so we can simplify 
```suggestion
  if (!vars.url) {
```